src/figuresClasses.ts

```diff
@@ -1,19 +1,112 @@
-export interface Figure {
 
+enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
+}
+
+type Color = 'red' | 'green' | 'blue';
+
+export interface Figure {
+  shape: Shape,
+  color: Color,
+  getArea(): number,
 }
 
-export class Triangle {
+export abstract class BaseFigure implements Figure {
+  abstract readonly shape: Shape;
 
+  abstract color: Color;
+
+  abstract getArea(): number;
 }
 
-export class Circle {
+export class Triangle extends BaseFigure {
+  readonly shape = Shape.Triangle;
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    super();
+
+    const arrayOfSides = [a, b, c];
+    const isSomeSideNegative:boolean = arrayOfSides.some(
+      (side: number) => (side < 0),
+    );
+
+    if (isSomeSideNegative) {
+      throw new Error('All sides should be positive!');
+    }
+
+    const [
+      sideA,
+      sideB,
+      sideC,
+    ]: number[] = arrayOfSides.sort((prev, curr) => prev - curr);
 
+    if (sideC >= sideA + sideB) {
+      throw new Error('Impossible to built a triangle with these sides!');
+    }
+  }
+
+  getArea(): number {
+    const semiperimeter = (this.a + this.b + this.c) / 2;
+
+    const area = Math.sqrt(
+      semiperimeter
+      * (semiperimeter - this.a)
+      * (semiperimeter - this.b)
+      * (semiperimeter - this.c),
+    );
+
+    return Number(area.toFixed(2));
```

Comment:
Why did you use `toFixed` here, but `math.floor(area * 100) / 100` in other classes? IMO it looks strange

Reply:
Better to use one approach and to create helper function for it